File: Test3.3/src/main/java/com/example/test_33/MainActivity.java
package com.example.test_33;

import androidx.appcompat.app.AppCompatActivity;

import android.graphics.Color;
import android.os.Bundle;
import android.view.ContextMenu;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private TextView textView;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        textView = (TextView) findViewById(R.id.txt);
        // 为文本框注册上下文菜单
        //registerForContextMenu(textView);
    }
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_main,menu);
        return true;
    }
    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.isCheckable()){
            //勾选菜单项
            item.setCheckable(true);
        }
        //switch 判断单击哪个菜单项，并有针对性的做出响应
        switch (item.getItemId()){
            case R.id.plain_item:
                Toast.makeText(this,"普通选项",Toast.LENGTH_LONG).show();
            case R.id.font_10:
                textView.setTextSize(10 * 2);
                break;
            case R.id.font_16:
                textView.setTextSize(16 * 2);
                break;
            case R.id.font_20:
                textView.setTextSize(20 * 2);
                break;
            case R.id.red_font:
                textView.setTextColor(Color.RED);
                break;
            case R.id.green_font:
                textView.setTextColor(Color.GREEN);
                break;
            case R.id.blue_font:
                textView.setTextColor(Color.BLUE);
                break;
        }
        return true;
    }
}

File: Test3.4/app/src/main/java/com/example/lab03_04/MainActivity.kt
package com.example.lab03_04

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.ActionMode
import android.view.Menu
import android.view.MenuInflater
import android.view.MenuItem
import android.widget.AbsListView
import android.widget.ArrayAdapter
import android.widget.ListView
import androidx.core.view.size
import kotlinx.android.synthetic.main.activity_main.*
import java.util.EnumSet.range

class MainActivity : AppCompatActivity() {

    private val itemList = ArrayList<item>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initItem()
        val adapter =BaseAdapter(this,R.layout.item_layout,itemList)
        listView.adapter =adapter
        with(listView){
            choiceMode = ListView.CHOICE_MODE_MULTIPLE_MODAL
            setMultiChoiceModeListener(object:AbsListView.MultiChoiceModeListener{
                var num = 0
                override fun onCreateActionMode(mode: ActionMode, menu: Menu?): Boolean {
                    val menuInflater : MenuInflater = mode.menuInflater
                    menuInflater.inflate(R.menu.actionmode_menu,menu)
                    num = 0
                    adapter.notifyDataSetChanged()
                    return true
                }

                override fun onPrepareActionMode(mode: ActionMode?, menu: Menu?): Boolean {
                    adapter.notifyDataSetChanged()
                    return false
                }

                public fun refresh(boolean: Boolean){
                    for (index in 0 until itemList.size){
                        itemList.get(index).Bo =boolean
                    }
                }

                override fun onActionItemClicked(mode: ActionMode?, item: MenuItem?): Boolean {
                    when(item?.itemId){
                        R.id.menu_all ->{
                            num = 0
                            var selectedNum = 0
                            for (index in 0 until itemList.size){
                                if (itemList.get(index).Bo){
                                    selectedNum++
                                }
                            }
                            if (selectedNum == itemList.size){
                                refresh(false)
                                num = 0
                            }else{
                                refresh(true)
                                num = itemList.size
                            }
                            adapter.notifyDataSetChanged()
                            if (mode != null) {
                                mode.title ="$num Selected"
                            }
                            return true
                        }
                        R.id.menu_delete ->{
                            var size =itemList.size-1
                            for(index in size downTo 0){
                                if (itemList.get(index).Bo){
                                    itemList.removeAt(index)
                                }
                            }
                            adapter.notifyDataSetChanged()
                            num = 0
                            refresh(false)
                            if (mode != null) {
                                mode.title ="$num Selected"
                            }
                            return true
                        }
                    }
                  return true
                }

                override fun onDestroyActionMode(mode: ActionMode?) {
                    refresh(false)
                    adapter.notifyDataSetChanged()
                }

                override fun onItemCheckedStateChanged(
                    mode: ActionMode?,
                    position: Int,
                    id: Long,
                    checked: Boolean
                ) {
                    if (itemList.get(position).Bo){
                        itemList.get(position).Bo=false
                        adapter.notifyDataSetChanged()
                        num--
                    }else {
                        itemList.get(position).Bo=true
                        adapter.notifyDataSetChanged()
                        num++
                    }
                    if (mode != null) {
                        mode.title = "$num Selected"
                    }
                }

            })
        }
    }

    fun initItem(){
        itemList.add(item("one"))
        itemList.add(item("two"))
        itemList.add(item("three"))
        itemList.add(item("four"))
        itemList.add(item("five"))
        itemList.add(item("six"))
        itemList.add(item("seven"))
        itemList.add(item("eight"))
        itemList.add(item("nine"))
        itemList.add(item("ten"))
    }
}